technology/kotlin-quarkus/src/main/kotlin/org/acme/kotlin/schooltimetabling/rest/TimetableDemoResource.kt

```diff
@@ -0,0 +1,447 @@
+package org.acme.kotlin.schooltimetabling.rest
+
+import jakarta.ws.rs.GET
+import jakarta.ws.rs.Path
+import jakarta.ws.rs.PathParam
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.Response
+import org.acme.kotlin.schooltimetabling.domain.Lesson
+import org.acme.kotlin.schooltimetabling.domain.Room
+import org.acme.kotlin.schooltimetabling.domain.Timeslot
+import org.acme.kotlin.schooltimetabling.domain.Timetable
+import org.eclipse.microprofile.openapi.annotations.Operation
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType
+import org.eclipse.microprofile.openapi.annotations.media.Content
+import org.eclipse.microprofile.openapi.annotations.media.Schema
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses
+import org.eclipse.microprofile.openapi.annotations.tags.Tag
+import java.time.DayOfWeek
+import java.time.LocalTime
+
+@Tag(name = "Demo data", description = "Timefold-provided demo school timetable data.")
+@Path("demo-data")
+class TimetableDemoResource {
+
+    enum class DemoData {
+        SMALL, LARGE
+    }
+
+    @APIResponses(
+        value = [APIResponse(
+            responseCode = "200", description = "List of demo data represented as IDs.", content = [Content(
+                mediaType = MediaType.APPLICATION_JSON,
+                schema = Schema(implementation = DemoData::class, type = SchemaType.ARRAY)
+            )]
+        )]
+    )
+    @Operation(summary = "List demo data.")
+    @GET
+    fun list(): Array<DemoData> {
+        return DemoData.entries.toTypedArray()
+    }
+
+    @APIResponses(
+        value = [APIResponse(
+            responseCode = "200", description = "Unsolved demo timetable.", content = [Content(
+                mediaType = MediaType.APPLICATION_JSON, schema = Schema(implementation = Timetable::class)
+            )]
+        )]
+    )
+    @Operation(summary = "Find an unsolved demo timetable by ID.")
+    @GET
+    @Path("/{demoDataId}")
+    fun generate(
+        @Parameter(
+            description = "Unique identifier of the demo data.", required = true
+        ) @PathParam("demoDataId") demoData: DemoData
+    ): Response {
+        val timeslotList: MutableList<Timeslot> = ArrayList(10)
+        var nextTimeslotId = 0L
+        timeslotList.add(
+            Timeslot(nextTimeslotId++, DayOfWeek.MONDAY, LocalTime.of(8, 30), LocalTime.of(9, 30))
+        )
```

Comment:
This is very verbose. Some later calls take even 5 lines.
Maybe by static importing `DayOfWeek` values, we can get this to fit on one line?